src/pokeService.js

```diff
@@ -17,6 +24,51 @@ async function getPokemonMoves(id) {
   return resolvePromises;
 }
 
+async function getPokemonEvoChain(id) {
+  const species = await getPokemonSpecies(id);
+  const evoChainUrl = species.data.evolution_chain.url;
+  const evoChains = await axios.get(evoChainUrl);
+  let evoData = evoChains.data.chain;
+
+  const evoChain = [];
+  do {
+    const numOfEvolutions = evoData.evolves_to.length;
+    let pokemon = await getPokemonByName(evoData.species.name);
+    evoChain.push({
+      url: evoData.species.url,
+      name: evoData.species.name,
+      id: pokemon.id,
+      image: pokemon.sprites.front_default,
+      min_level: !evoData.evolves_to[0] ? 1 : evoData.evolves_to[0].evolution_details[0].min_level,
```

Comment:
This line is very unreadable at this point I would suggest you put this logic in a function